homeassistant/components/device_tracker/volvooncall.py

```diff
@@ -8,32 +8,33 @@
 
 from homeassistant.util import slugify
 from homeassistant.helpers.dispatcher import (
-    dispatcher_connect, dispatcher_send)
+    async_dispatcher_connect, async_dispatcher_send)
 from homeassistant.components.volvooncall import DATA_KEY, SIGNAL_VEHICLE_SEEN
 
 _LOGGER = logging.getLogger(__name__)
 
 
-def setup_scanner(hass, config, see, discovery_info=None):
+async def async_setup_scanner(hass, config, async_see, discovery_info=None):
     """Set up the Volvo tracker."""
     if discovery_info is None:
         return
 
-    vin, _ = discovery_info
+    vin, component, attr = discovery_info
     voc = hass.data[DATA_KEY]
-    vehicle = voc.vehicles[vin]
+    instrument = voc.instrument(vin, component, attr)
 
-    def see_vehicle(vehicle):
+    async def see_vehicle(vehicle):
         """Handle the reporting of the vehicle position."""
-        host_name = voc.vehicle_name(vehicle)
+        if vehicle != vin:
```

Comment:
Create a vehicle specific signal to connect the dispatcher target to instead.

Reply:
✓ Re-using the generic signal for state update introduced above instead.